include/Model/Abstract_Languages_Proxy.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\Model;
+
+use PLL_Language;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Abstract class allowing to proxy the list of languages.
+ *
+ * @since 3.8
+ */
+abstract class Abstract_Languages_Proxy {
+	/**
+	 * @var Languages
+	 */
+	protected $languages;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.8
+	 *
+	 * @param Languages $languages The object to proxy.
+	 */
+	public function __construct( Languages $languages ) {
+		$this->languages = $languages;
+	}
+
+	/**
+	 * Returns the proxy's key.
+	 *
+	 * @since 3.8
+	 *
+	 * @return string
+	 *
+	 * @phpstan-return non-falsy-string
+	 */
+	abstract public static function key(): string;
```

Comment:
Not sure we need it to be static.

Reply:
I fear it will come back and bite our asses soon or later but done in [3c4d854](https://github.com/polylang/polylang/pull/1727/commits/3c4d85420f637ed40460a1819e499b9ab3cd5a08).